test/functional/tests/fault_injection/test_fault_injection_interrupts.py

```diff
@@ -263,24 +270,26 @@ def test_interrupt_core_remove(cache_mode, filesystem):
             core_part.mount(mount_point)
 
         with TestRun.step("Check md5 sum of test file again."):
-            if test_file_md5sum_before != test_file.md5sum():
+            is_sum_diff_after_interrupt = test_file_md5sum_interrupt != test_file.md5sum()
+            if test_file_md5sum_before != test_file.md5sum() or is_sum_diff_after_interrupt:
```

Comment:
How about splitting it into two separate error messages? It might be valuable to know at which point the corruption occurred